pkg/kvstore/etcd.go

```diff
@@ -371,7 +371,10 @@ type etcdClient struct {
 
 	extraOptions *ExtraOptions
 
-	limiter       *ciliumrate.APILimiter
+	// limiterLock protects limiter from concurrent access
+	limiterLock lock.RWMutex
+	limiter     *ciliumrate.APILimiter
```

Comment:
what about an `atomic.Pointer[ciliumrate.APILimiter]`? Might be easier to reason about.